packages/flutter/lib/src/rendering/layer.dart

```diff
@@ -679,6 +765,20 @@ class ContainerLayer extends Layer {
     assert(transform != null);
   }
 
+  /// Returns the descendants of this layer in depth first order.
+  Iterable<Layer> depthFirstIterateChildren() sync* {
```

Comment:
In general sync* has very bad performance, it is better to avoid it.

Reply:
Is there some way to inspect the generated  code here?  I strongly suspect it will be at least as efficient as writing our own iterable implementation for this in the current usage...

Reply:
Although this is making me realize that I'm modifying the tree while iterating it which is a bad idea.  I can fix that though.